### PR TITLE
2058 rql select

### PR DESCRIPTION
**Pull request by mrix**
I tried to use `select` in MongoDb query but it cannot be applied to references and throws exceptions if invalid fields is specified.
Also I tried to implement `ExclusionStrategy` for `Serializer` but it cannot be applied to `schemaless` (`hash` or `array`) fields.
Then I implemented  `ResponseListener` that filters JSON using `ObjectSlicer`

### CODE REVIEW

File: src/Graviton/RestBundle/Tests/Utils/ObjectSlicerTest.php [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+            'slice deep' => [
+                (object) ['a' => (object) ['b' => 2], 'c' => 3],
+                'a.b',
+                (object) ['a' => (object) ['b' => 2]],
+            ],
+
+            'unexist' => [
+                (object) ['a' => (object) ['b' => 2]],
+                'a.b.c',
+                (object) ['a' => (object) []],
```
Comment (hairmare) [inline src/Graviton/RestBundle/Tests/Utils/ObjectSlicerTest.php]: Why would this only return ``(object) ['a' => (object) []]``? Looking at how "slice deep" behaves I'd probably expect the return value to contain ``b``s value. Wouldn't it be more intuitive if it behaved more like "deep array slice"? Probably not, but could you please clarify this?

Comment (mrix) [inline src/Graviton/RestBundle/Tests/Utils/ObjectSlicerTest.php]: I implemented MongoDb behavior:
```
> db.Test.insert({a: {b: 2}})
WriteResult({ "nInserted" : 1 })
> db.Test.findOne()
{ "_id" : ObjectId("55b9049cef8145bb7b1462c4"), "a" : { "b" : 2 } }
> db.Test.findOne({}, {"a.b.c": 1})
{ "_id" : ObjectId("55b9049cef8145bb7b1462c4"), "a" : { } }
```

Comment (hairmare) [inline src/Graviton/RestBundle/Tests/Utils/ObjectSlicerTest.php]: Thanks make sense, thanks

File: src/Graviton/RestBundle/Utils/ObjectSlicer.php [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                if (!property_exists($dst, $key)) {
+                    $dst->{$key} = $value;
+                } elseif (is_object($value) || is_array($value)) {
+                    $dst->{$key} = $this->mergeSliceRecursive($dst->{$key}, $value);
+                }
+            }
+        } else {
+            foreach ($src as $key => $value) {
+                if (is_object($value) || is_array($value)) {
+                    $a[$key] = $this->mergeSliceRecursive($dst[$key], $value);
```
Comment (hairmare) [inline src/Graviton/RestBundle/Utils/ObjectSlicer.php]: Please initialize ``$a`` before using it (scrutinizer found [this](https://scrutinizer-ci.com/g/libgraviton/graviton/inspections/bc3a523d-b62a-4826-b3c0-419df2450318/issues/files/src/Graviton/RestBundle/Utils/ObjectSlicer.php?status=new&orderField=path&order=asc) and we prefer always initializing vars anyway)

Comment (mrix) [inline src/Graviton/RestBundle/Utils/ObjectSlicer.php]: Fixed. Thanks @hairmare.

**Pull request merged**